src/arrayMethodJoin.js

```diff
@@ -4,8 +4,33 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const mainArrayClone = [];
+    let mainSeparatorClone = separator;
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (separator === null) {
+      mainSeparatorClone = 'null';
+    }
```

Comment:
why would you need this, I guess if there's null it will just be ','

Reply:
I need it for this test:
![null_test](https://github.com/mate-academy/js_array-method-join/assets/28891202/4cef19b0-9562-4000-8d1e-da8745ff0604)